creator-node/src/routes/prometheusMetricsRoutes.js

```diff
@@ -0,0 +1,12 @@
+module.exports = function (app) {
+  app.get('/prometheus_metrics', async (req, res) => {
+    const prometheusRegistry = req.app.get('serviceRegistry').prometheusRegistry
+
+    try {
+      res.setHeader('Content-Type', prometheusRegistry.register.contentType)
```

Comment:
This is when we would ideally collect gauge metrics. See [my PR](https://github.com/AudiusProject/audius-protocol/pull/2888/files#diff-eef81b1feadb8789a757267aae588591be3363ec1c8f525651708d9fe89e95f9R16-R18).

When you're creating the gauges code, think about exporting this (or a similar metric):

"What is the current uptime of the server?"

Note that the answer should be current as of scrape-time.

Reply:
Oh cool! Glad to see they made "collecting" gauges more accessible! Let's definitely use their preferred way.

We want gauges at scape time depending on the questions we're asking. A good example is "What is the current uptime of the server?" That question could only really be answered at scrape time, but glad we have the new `collect` option for when those cases arise.

And major +1 on being cautious of expensive operations at scrape time.

Reply:
i dont think i really follow this - how would a gauge measure "current uptime of server", and why's that something that can only be answered at scrape time vs recorded just before

Reply:
Perhaps this helps?

Let's take the following pseudocode:

```
uptime = os.process.uptime()
metric.save(uptime)
```

Let's say `uptime` and `metric` resolve as expected, now the issue becomes where would you place this code (or equivalent in Node) in our current creator-node codebase?

We wouldn't want it on the init path since it would only get called once at startup. We could bury it in a worker and as long as the worker has something to consume, we *should* get our metrics. We could bury it on our request path, but if we stop getting web requests, we won't have this metric being exported.

If we choose any of the above locations, we run the risk of no longer updating the "current" uptime within Prometheus and our Grafana datapoint will look like a stale uptime is an accurate value.

So the question becomes, given this ideal metric, what file/line would you add this gauge into creator-node with a strong consideration towards accurate, up-to-date information?

Perhaps there's a thread-loop specific to Node that's stuck in a while loop, or something I'm unaware of, and we can place it there instead of populating the metric at scrape time?

Reply:
Additional constraint for the above code/line location finding thought experiment:

* https://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics

Reply:
synced on call - we're on same page, the correct way to do this would be to add this to our `MonitoringQueue` which runs on a cron, and instead of populating custom metrics in redis we would populate prometheus metrics

great call